sound/soc/sof/intel/hda-dai.c

```diff
@@ -311,6 +342,12 @@ static int hda_link_pcm_trigger(struct snd_pcm_substream *substream,
 	hda_stream = hstream_to_sof_hda_stream(link_dev);
 
 	dev_dbg(dai->dev, "In %s cmd=%d\n", __func__, cmd);
+
+	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK)
+		w = dai->playback_widget;
+	else
+		w = dai->capture_widget;
+
```

Comment:
can you add this commit before

https://github.com/plbossart/sound/commit/a564e5ee3663e92095a3ae596a647ec9a077bae4

this will make things simpler for later merges.

Reply:
the comment to use snd_soc_dai_get_widget() is still valid.

Reply:
fixed now